WordPress/src/main/java/org/wordpress/android/ui/reader/views/compose/tagsfeed/ReaderTagsFeed.kt

```diff
@@ -145,8 +148,76 @@ private fun Loading() {
 }
 
 @Composable
-private fun Empty() {
-// TODO empty state (https://github.com/wordpress-mobile/WordPress-Android/issues/20584)
+private fun Empty(uiState: UiState.Empty) {
+    Column(
+        modifier = Modifier.fillMaxSize(),
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        // Title
+        Text(
+            modifier = Modifier
+                .padding(
+                    start = Margin.ExtraExtraMediumLarge.value,
+                    end = Margin.ExtraExtraMediumLarge.value,
+                    bottom = Margin.Medium.value,
+                ),
+            text = stringResource(id = R.string.reader_discover_empty_title),
+            textAlign = TextAlign.Center,
+            fontSize = 20.sp,
+            style = MaterialTheme.typography.subtitle1,
+            color = MaterialTheme.colors.onSurface.copy(
+                alpha = ContentAlpha.medium,
+            ),
+        )
+        // Subtitle
+        Text(
+            modifier = Modifier
+                .padding(
+                    start = Margin.ExtraExtraMediumLarge.value,
+                    end = Margin.ExtraExtraMediumLarge.value,
+                    bottom = Margin.Large.value,
+                ),
+            text = stringResource(id = R.string.reader_discover_empty_subtitle_follow),
+            textAlign = TextAlign.Center,
+            style = MaterialTheme.typography.subtitle1,
+            color = MaterialTheme.colors.onSurface.copy(
+                alpha = ContentAlpha.medium,
+            ),
+        )
+        // Button
+        Button(
+            onClick = uiState.onOpenTagsListClick,
+            modifier = Modifier.padding(
+                start = Margin.ExtraMediumLarge.value,
+                end = Margin.ExtraMediumLarge.value,
+                bottom = Margin.ExtraLarge.value,
+            ),
+            contentPadding = PaddingValues(
+                horizontal = 32.dp,
+                vertical = 8.dp,
+            ),
+            elevation = ButtonDefaults.elevation(
+                defaultElevation = 0.dp,
+                pressedElevation = 0.dp,
+            ),
+            colors = ButtonDefaults.buttonColors(
+                contentColor = MaterialTheme.colors.onPrimary,
+                backgroundColor = MaterialTheme.colors.onSurface,
+            ),
+        ) {
+            Row() {
```

Comment:
Is the Row really required here? If no additional components are planned to be added alongside the text, the Row could be removed.

Reply:
Thanks! It wasn't needed. I've removed it.